src/rscore-pp-block-factory.cpp

```diff
@@ -0,0 +1,53 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "rscore-pp-block-factory.h"
+
+#include "proc/decimation-filter.h"
+#include "proc/disparity-transform.h"
+#include "proc/hdr-merge.h"
+#include "proc/hole-filling-filter.h"
+#include "proc/sequence-id-filter.h"
+#include "proc/spatial-filter.h"
+#include "proc/temporal-filter.h"
+#include "proc/threshold.h"
+
+#include <rsutils/string/nocase.h>
+#include <rsutils/json.h>
+
+
+namespace librealsense {
+
+
+std::shared_ptr< processing_block_interface >
+rscore_pp_block_factory::create_pp_block( std::string const & name, nlohmann::json const & settings )
+{
+    // These filters do not accept settings (nor are settings recorded in ros_writer)
+    (void *)&settings;
+
+    if( rsutils::string::nocase_equal( name, "Decimation Filter" ) )
```

Comment:
How is this code better than before?
Performance is worse (searching in json)

Magic strings are used (no header that group all)

Reply:
The code compared strings before, too.

Yes, before they were "hard-coded" in that they were generated from the extension names. But we don't want to use the extension names. If you prefer, I can compare block names and fall back to "automatic" extension name comparison which was case-sensitive, but I don't think that's readable or easy to understand at all.

This is the factory. It centralizes the knowledge of processing blocks, and lets us add processing blocks dynamically in different modules. I like this much better than having extension names spread all over the place, many times not in any human-preferable way ("Hdr Merge" rather than "HDR Merge").

I can change to fit your preference, but the factory idea is much better.

Reply:
As far as performance, I can change to a map if you like. But we compared all strings before, too. This happens once during device creation.